src/codegate/pipeline/cli/cli.py

```diff
@@ -94,6 +94,21 @@ def _get_cli_from_continue(last_user_message_str: str) -> Optional[re.Match[str]
 
     return codegate_regex.match(last_user_message_str)
 
+def _get_cli_from_copilot(last_user_message_str: str) -> Optional[re.Match[str]]:
+    """
+    Process Copilot-specific CLI command format.
+
+    Copilot sends messages in the format:
+    <attachment>file contents</attachment>codegate command
+
+    Args:
+        last_user_message_str (str): The message string from Copilot
+
+    Returns:
+        Optional[re.Match[str]]: A regex match object if command is found, None otherwise
+    """
+    cleaned_text = re.sub(r'<attachment>.*</attachment>', '', last_user_message_str, flags=re.DOTALL)
```

Comment:
optimization: pre-declare and pre-compile each regular expression in the constructor or even globally so this function would only need to evaluate the regex as opposed to compile+evaluate.

Reply:
let's do it separately for all the regexes here. We should also move the client helpers to the client interface.